library/Control/Monad/AWS/EnvT.hs

```diff
@@ -47,6 +48,7 @@ newtype EnvT m a = EnvT
     , MonadUnliftIO
     , MonadResource
     , MonadReader Env
+    , MonadError e
```

Comment:
Nice! I feel like we should add `MonadState` too. I'm half-inclined to do all the type-classes from `mtl`: https://hackage.haskell.org/package/mtl-2.3.1. We've already got the dependency right? But maybe just starting with `MonadState` (so you'd now cover the super common reader, error, state ones).

`MonadLogger` is an interesting case. It would mean another dependency, but may be worth it as I feel wanting to log from within `EnvT` is extremely likely.

Tough to decide where to draw the line.

Reply:
Can I suggest making this a part of a kind of "renovation"? Upgrading stuff in one PR, adding as many mtl-style type class instances in another. WDYT?

Reply:
> Can I suggest making this a part of a kind of "renovation"?

What do you mean by "this" here?

Reply:
I mean adding more instances. The "rennovation" I'm thinking of is two PRs, one to update the LTS resolver version, another to incorporate additional instances. Does that answer your question?

Reply:
Oh OK. I'm typically all about isolated PRs, but in this case I just don't see why you'd add some mtl instance now and save some other mtl instances for later. If you plan to do them, why not do them together?

I Approved, so do what you like, I just don't really follow that particular reasoning.

Reply:
Oh, I don't mean in this PR, I would close this PR in favor of the two I mentioned

Reply:
Haha OK, well then now what I don't get why you would rather close this and open 2 new PRs, instead of letting this just turn into one of those 2 you described. But again, you do you :)

Reply:
Done, it's all in this PR now